go/vt/vtgate/tabletgateway_test.go

```diff
@@ -84,7 +84,8 @@ func TestTabletGatewayBeginExecute(t *testing.T) {
 
 func TestTabletGatewayShuffleTablets(t *testing.T) {
 	hc := discovery.NewFakeHealthCheck(nil)
-	tg := NewTabletGateway(context.Background(), hc, nil, "local")
+	ts := &fakeTopoServer{}
```

Comment:
Earlier implementation used to pass nil as the topo server and through some side effect it worked. However after removing the health check code this no longer worked. Hence here, and in other places below, we are passing an instance of the test (fake) topo server.